modules/users.js

```diff
@@ -0,0 +1,143 @@
+import { API } from '../utils';
+import { SET_VISIBLE, SET_SEARCH_USER } from './userEntry';
+
+export const SET_USERS = 'users/SET_USERS';
+export const SET_FETCH = 'users/SET_FETCH';
+
+const initialState = {
+  all: [],
+  current: [],
+  page: 0,
+  isFetched: false,
+  params: {},
+  filters: {},
+};
+
+const format = (users) => {
+  return users.map((user) => ({
+    ...user,
+    fullname: `${user.firstname} ${user.lastname}`,
+    tournamentName: user.team ? user.team.tournament.shortName : '',
+    teamName: user.team ? user.team.name : (user.type === 'visitor' ? '(visiteur)' : ''),
+    paidLabel: user.isPaid ? '✔' : '✖',
+    scanned: user.scanned ? '✔' : '✖',
+    permissionsLabel: user.permissions || '',
+  }));
+};
+
+export default (state = initialState, action) => {
+  switch (action.type) {
+    case SET_USERS:
+      return {
+        ...state,
+        all: action.all,
+      };
+    case SET_FETCH:
+      return {
+        ...state,
+        ...action,
+      };
+    default:
+      return state;
+  }
+};
+
+export const fetchUsers = () => async (dispatch, getState) => {
+  const page = getState().users.page;
+  const res = await API.get(`admin/users?page=${page}`);
+  const formatUsers = format(res.data.users);
+  dispatch({
+    type: SET_FETCH,
+    current: formatUsers,
+    all: formatUsers,

```

Comment:
```suggestion
    all: [formatUsers],
```
C'est pas une erreur! c'est voulu